Cockpit 310 (#704)
cockpit-project/cockpit-project.github.io: _posts/2024-01-31-cockpit-310.md

---
title: Cockpit 310
author: mvo
date: '2024-01-31'
tags: cockpit, machines
slug: cockpit-310
category: release

> summary: ''

Reviewer: ```suggestion
summary: 'Btrfs creation, swap device support, virtual machine SPICE migration, and external VM snapshots'
```

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 310 and cockpit-machines 306:

## Storage: More support for btrfs

Cockpit now understands btrfs filesystems even better. It can create new btrfs
filesystems and new subvolumes.

## Storage: improved support for swap

Cockpit can now format block devices as swap and will maintain a fstab entry for them.

## Machines: Action to Replace SPICE devices

There are some VM host machines which don't support [SPICE
features](https://www.spice-space.org/spice-user-manual.html). In
particular, Red Hat Enterprise Linux 9 does not support SPICE at all,
and on other operating systems the SPICE support packages may not be
installed.

> The VM action menu got a "Replace SPICE support" action which prepares
> a VM so that upgrading/changing the host or migrating to a host
> without SPICE support will work without failing to start the VM
> afterwards. In particular, it converts SPICE graphics to VNC, "qxl"
> video to "vga", and drops all other spice devices which don't have a
> replacement (audio and USB passthrough).

Reviewer: ```suggestion
Virtual machines that use SPICE now have a "Replace SPICE support" action, which is useful for adjusting machines on systems that do not have SPICE support, or preparing a VM to be ready to run on a new host after migration. Replacing SPICE converts SPICE graphics to VNC, "qxl" video to "vga", and drops spice devices which do not have a replacement, such as audio and USB passthrough.

This action can be selected from the VM's menu on the machines list.
```

![screenshot of action to replace spice devices]({{ site.baseurl }}/images/310-machines-action-to-replace-spice-devices.png)

> If the host already doesn't support SPICE (any more), but the
> currently shown VM still uses SPICE, Cockpit shows a notification to
> explain how to fix it, which will lead to the same dialog:

Reviewer: ```suggestion
When a host does not have SPICE support, but the VM is configured for SPICE, then Cockpit will show an alert to explain the issue and provide a way to run the "replace SPICE" action. This may occur if a machine was migrated without replacing SPICE or if the host system was upgraded to a new version of a distribution that no longer supports SPICE.
```

![screenshot of action to replace spice devices]({{ site.baseurl }}/images/310-machines-action-to-replace-spice-devices-2.png)

> ## Machines: Create external snapshots when supported

Reviewer: ```suggestion
## Machines: External snapshot support
```

> "External" VM snapshots are more reliable and flexible than the "internal" format. They also work for raw disk images, not just for qcow2. In addition, the internal format was [declared deprecated in RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.4_release_notes/deprecated_functionality#deprecated-functionality_virtualization).

Reviewer: ```suggestion
"External" VM snapshots are more reliable and flexible than the "internal" format. External snapshots also work for raw disk images, not just for qcow2. Additionally, the internal format was [declared deprecated in RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.4_release_notes/deprecated_functionality#deprecated-functionality_virtualization).
```

> Creating snapshots in cockpit-machines now uses the external format with libvirt 9.9.0 or later,  which is currently available in Fedora Rawhide, CentOS 9 Stream, Arch Linux and Debian unstable/testing. If the VM is running, you can customize the location of the memory snapshot file:

Reviewer: ```suggestion
Cockpit Machines uses the external snapshot format when using libvirt 9.9.0 or later. This is currently available in Fedora Rawhide, CentOS 9 Stream, Arch Linux, Debian Unstable, and Debian Testing.

The memory snapshot file location can be changed while the VM is running:
```
